walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java

```diff
@@ -21,7 +21,18 @@
   /**
    * Returns a greeting for the given name.
    */
+  // default trimmer
   public String greet(String name) {
+    return "Hello " + name.trim().replaceAll("[@#$%!? ]", "");
+  }
+
+  // custom trimmer
+  private static final String[] INVALID_ARRAY = {" ", "@", "%", "!"};
+
+  public String greetFromSpecificList(String name) {
+    for (String c : INVALID_ARRAY) {
+      name = name.replaceAll(c, "");
```

Comment:
If you're not familiar with Big-O complexities, we can discuss this more in one of our stand-ups.

I suspect that replaceAll runs in O(N) time. However we need to call replaceAll for however many elements we have in INVALID_ARRAY. So if we have M elements, we're looking at O(N*M) time. I'm wondering if we could do better than that by concatenating INVALID_ARRAY into a single string and then calling replaceAll just once. 

What do you think?